doc: Update known issues for NCS v2.9.0 (#19231)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

Amazon Sidewalk
===============

.. rst-class:: v2-9-0 v2-8-0

KRKNWK-19647: nRF52 has high power consumption in sleep mode for sub-GHz sample
  nRF52840 device experiences high current consumption in sleep mode with Bluetooth® LE link when using sub-GHz Amazon libraries.
  This issue does not occur with Bluetooth LE-only configurations.

  **Affected platforms:** nRF52840

.. rst-class:: v2-9-0 v2-8-0

KRKNWK-19647: nRF53 has high power consumption in sleep mode
  The nRF5340 device has high current consumption in sleep mode across all link configurations.

  **Affected platforms:** nRF5340

.. rst-class:: v2-9-0 v2-8-0

> KRKNWK-19459: AES crypt error occurs after a factory reset

Reviewer: @divipillai All Amazon Sidewalk known issues are now resolved except for this one (KRKNWK-19459)

Author: Removed v2-9-0 for all entries except for KRKNWK-19459.

Bluetooth LE
============

.. rst-class:: v2-9-0 v2-8-0

DRGN-23363: A flash operation executed on the system workqueue might result in ``-ETIMEDOUT``
  A flash operation executed on the system workqueue might result in ``-ETIMEDOUT`` if there is an active Bluetooth LE connection.

>   **Workaround:** Move flash operations to a different workqueue.

Reviewer 2: True, `different thread` is better, thanks for the suggestion.

> btw. Isn't this issue related to `NCSDK-29354` ? (seems a bit like a duplicate)

Yeah, but according to the reason in `NCSDK-29354`, these are 2 different issues which result in the same behavior. I was able to reproduce the timeout even with `CONFIG_BT_CONN_TX_NOTIFY_WQ` enabled, which doesn't exclude the fact that the issue explained in `NCSDK-29354` `happens because Bluetooth RX context waits for the connection TX notify that is done in the system workqueue context`.

I think I'd keep both known issues.

.. rst-class:: v2-8-0 v2-7-0

NCSDK-30458: The Bluetooth LE applications built with the ``nordic-bt-rpc`` snippet do not work on nRF54H20 devices

>   When using the ``nordic-bt-rpc`` sysbuild snippet for building Bluetooth LE applications for nRF54H20 in the Bluetooth RPC configuration, the build system creates an invalid memory map for the application core, resulting in a runtime failure.

Reviewer 3: ```suggestion
  When using the ``nordic-bt-rpc`` sysbuild snippet for building Bluetooth LE applications for an nRF54H20 device in the Bluetooth RPC configuration, the build system creates an invalid memory map for the application core, resulting in a runtime failure.
```

The issues in this section are related to the :ref:`ug_bt_mesh` protocol.

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Reviewer 4: ```suggestion
.. rst-class:: v2-8-0 v2-7-0
```
Resloved

The issues in this section are related to the :ref:`ug_matter` protocol.

.. rst-class:: v2-9-0 v2-8-0

KRKNWK-19388: The smart plug functionality of Matter Bridge application does not work with Apple Home application
  The additional functionality of a smart plug that runs alongside the Matter Bridge functionality is not visible in the Apple Home application.
  It applies to every functionality reflected by the static Matter endpoint and run alongside dynamic endpoints represented by the Matter Bridge.

.. rst-class:: v2-7-0 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-4 v2-4-3 v2-4-2 v2-4-1 v2-4-0

KRKNWK-19524: Reliability issues with multicast message delivery for certain Access Points

>   Certain Access Points require Wi-Fi® devices to send Multicast Listener Discovery version 2 (MLDv2) Report messages for multicast packets to be properly delivered to a given address.

Reviewer 3: Access Points -> access points?

The issues in this section are related to the :ref:`ug_wifi` protocol.

.. rst-class:: v2-9-0 v2-8-0

SHEL-3345 : Wi-Fi Coex (Bluetooth LE and Thread) is broken

>   The Wi-Fi Coex is brokn for Bluetooth LE and Thread due to missing devicetree node in ``cpuapp``.

Reviewer 3: ```suggestion
  The Wi-Fi Coex is broken for Bluetooth LE and Thread due to missing devicetree node in ``cpuapp``.
```

The issues in this section are related to the :ref:`ug_wifi` protocol.

.. rst-class:: v2-9-0 v2-8-0

NCSDK-29650: Wi-Fi connection times are increased in ``_ns`` builds

>   The latency has increased signficantly in latest main branch.

Reviewer 5: ```suggestion
  The latency has increased significantly in latest main branch.
```

  #. To increase the scheduler queue size, replace ``XYZ`` next to ``ZB_CONFIG_SCHEDULER_Q_SIZE`` with the value of your choice, ranging from ``48U`` to ``256U``.
  #. To increase the buffer pool size, replace ``XYZ`` next to ``ZB_CONFIG_IOBUF_POOL_SIZE`` with the value of your choice, ranging from ``48U`` to ``127U``.

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-4 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0

Reviewer 6: This one is fixed by https://github.com/nrfconnect/sdk-nrf/pull/19431

The issues in this section are related to the :ref:`asset_tracker_v2` application.

.. rst-class:: v2-9-0 v2-8-0

NCSDK-30983: :ref:`asset_tracker_v2` with :file:`overlay-lwm2m` (LwM2M overlay) reboots after GNSS fix
  This issue affects Asset Tracker v2 with the LwM2M overlay.

>   After GNSS fix, the device reboots due to a NULL pointer exception.

Reviewer 3: ```suggestion
  After a GNSS fix, the device reboots due to a NULL pointer exception.
```

  **Affected platforms:** Thingy:91

>   **Workaround:** Set the :ref:`CONFIG_SLM_UART_RX_BUF_SIZE <CONFIG_SLM_UART_RX_BUF_SIZE>` Kconfig option to ``2048`` to ensure that there is adequate buffer space to receive traffic from the Connectivity bridge and disable the hw-flow-control from :file:`boards\thingy91_nrf9160_ns.overlay`.

Reviewer 3: ```suggestion
  **Workaround:** Set the :ref:`CONFIG_SLM_UART_RX_BUF_SIZE <CONFIG_SLM_UART_RX_BUF_SIZE>` Kconfig option to ``2048`` to ensure that there is adequate buffer space to receive traffic from the Connectivity bridge and disable the ``hw-flow-control`` from :file:`boards\thingy91_nrf9160_ns.overlay`.
```
Not so sure about this.

Author: Since this is in the code, I think double backtick should be used. Updated.

>   If even larger buffer spaces are required, the :kconfig:option:`CONFIG_BRIDGE_BUF_SIZE` Kconfig option for Connectivity bridge and the :ref:`CONFIG_SLM_UART_RX_BUF_SIZE <CONFIG_SLM_UART_RX_BUF_SIZE>` Kconfig option for SLM, must be set accordingly.

Reviewer 3: ```suggestion
  If even larger buffer spaces are required, set the :kconfig:option:`CONFIG_BRIDGE_BUF_SIZE` Kconfig option for Connectivity bridge and the :ref:`CONFIG_SLM_UART_RX_BUF_SIZE <CONFIG_SLM_UART_RX_BUF_SIZE>` Kconfig option for SLM accordingly.
```

    nRF Desktop is also affected by :ref:`hogp_readme` library's issue :ref:`NCSDK-30288 <ncsdk_30288>`.
    The issue may cause accessing data under ``NULL`` pointer in case of Bluetooth disconnection while forwarding a configuration channel operation in the :ref:`nrf_desktop_hid_forward`.

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Reviewer 7: works fine now

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Reviewer 8: Do you recall since when it's fixed?

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Author: @zycz : Please confirm the versions this issue is fixed and if the issue is actually related to factory reset (as described in the workaround)?

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Reviewer 7: It was fixed in late September with secdom binary update.
Isn't factory reset also a software reset? I don't see any contradiction in what is written there.

  **Workaround:** Press the **RESET** button on the nRF54H20 DK after performing a factory or software reset of the device.

.. rst-class:: v2-8-0

> NCSDK-30261: Bluetooth LE connection parameter results in a loop in the :ref:`nrf_desktop_ble_conn_params`

Reviewer 8: `Bluetooth LE connection parameter update is triggered multiple times in the :ref:`nrf_desktop_ble_conn_params``

  **Affected platforms:** nRF5340 Audio DK

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Reviewer 9: This was introduced in NCS 2.5.0.

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0

Author: Updated.

  **Affected platforms:** nRF5340 Audio DK

.. rst-class:: v2-9-0 v2-8-0 v2-7-0

OCT-3006: Setting a custom bitrate using :kconfig:option:`CONFIG_BT_AUDIO_BITRATE_UNICAST_SINK` will have no effect
  This is because the application reverts to one of the BAP presets.

  **Affected platforms:** nRF5340 Audio DK

.. rst-class:: v2-9-0 v2-8-0 v2-7-0 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0

OCT-3248: A race condition between capturing timers and the RTC tick resetting the HF timer

>   This issue might cause time to jump ahead by about 30 us.

Reviewer 3: ```suggestion
  This issue might cause time to jump ahead by about 30 µs.
```

Bluetooth samples
=================

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0 v2-6-2 v2-6-1 v2-6-0

Reviewer 10: This was fixed by @maje-emb in the following PR:

https://github.com/nrfconnect/sdk-nrf/pull/15960

NCS 2.7.0, 2.8.0 and 2.9.0 should be removed.

@maje-emb, it would be good to get confirmation from you

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0 v2-6-2 v2-6-1 v2-6-0

Reviewer 11: @maje-emb is not available today, but I confirm this is fixed - `2.9.0` to be removed.

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0 v2-6-2 v2-6-1 v2-6-0

Author: As per above comment, I have only removed the 2.9.0 version.

The issues in this section are related to :ref:`matter_samples`.

> .. rst-class:: v2-9-0 v2-8-0

Reviewer 12: This is not present in 2.9.0 anymore

.. rst-class:: v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-3-1 v1-2-1 v1-2-0

NCSDK-12912: The :ref:`liblwm2m_carrier_readme` library does not recover if initial network connection fails

>   When the device is switched on, if :c:func:`lte_lc_connect` returns an error at timeout, it will cause ``lwm2m_carrier_init()`` to fail.

Reviewer 3: ```suggestion
  When the device is switched on, if the :c:func:`lte_lc_connect` function returns an error at timeout, it will cause ``lwm2m_carrier_init()`` to fail.
```

> The issues in this section are related to :ref:`app_build_system`.

Reviewer 11: Please add a new issue per https://github.com/nrfconnect/sdk-nrf/pull/19292
.. rst-class:: v2-8-0 v2-7-0

NCSDK-30458: The BLE applications build with the `nordic-bt-rpc` snippet don't work on nRF54H20 devices.
  When using the ``nordic-bt-rpc`` sysbuild snippet for building BLE applications for nRF54H20 in the Bluetooth RPC configturation, the build system creates an invalid memory map for the application core resulting in a runtime failure.

**Workaround:** Add the `status = "disabled";` line in the `cpurad-rw-partitions` node definition in the :file:`snippets/nordic-bt-rpc/boards/nrf54h20dk_nrf54h20-mem-map-move.dtsi` file.

Author: Updated.

  You can `download this file <nRF52820 CMakeLists.txt_>`_ from the upstream Zephyr repository.
  After you add it, the file is automatically included by the build system.

> .. rst-class:: v2-9-0 v2-8-0 v2-7-0 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-4 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

Reviewer 13: resolved for 2.9.0, expectation for support for  SB_SIGNING_KEY_FILE environment variable  was removed.

.. rst-class:: v2-5-3 v2-5-2 v2-5-1 v2-5-0

NCSDK-24203: If fault injection hardening (FIH) is enabled, a bug is observed in the :c:func:`boot_image_check_hook` function

>   Due to this, multicore applications cannot be booted for nRF5340 MCUboot builds with simultaneous mult-image update enabled.

Reviewer 3: ```suggestion
  Due to this, multicore applications cannot be booted for nRF5340 MCUboot builds with simultaneous multi-image update enabled.
```

The issues in this section are related to :ref:`MCUboot <mcuboot_wrapper>`.

> .. rst-class:: v2-9-0 v2-8-0

Reviewer 13: .. rst-class:: v2-8-0

The issues in this section are related to :ref:`MCUboot <mcuboot_wrapper>`.

.. rst-class:: v2-9-0 v2-8-0

NCSDK-30867: KMU keys are not protected against being revoked by the application that is running in secure mode
  You must provision *locked* keys if you cannot trust the application running in secure mode.

>   This is valid for applications that miss TEE setup to consider the security aspect of the application.

Reviewer 13: ```suggestion
  This is valid for applications that is considered as cannot to be trusted on relevant level.
```

>   **Affected platforms:** nRF54L15

Reviewer 13: l05 an l10 as well

The issues in this section are related to :ref:`nrfxlib:mpsl`.

.. rst-class:: v2-9-0

> UARTE will have a frequency error beyond the specific limits

Reviewer 3: Do we have a Jira issue?

.. rst-class:: v1-4-2 v1-4-1

DRGN-15176: :kconfig:option:`CONFIG_SYSTEM_CLOCK_NO_WAIT` is ignored when Low Frequency Clock is started before initializing MPSL

>   If the application starts the Low Frequency Clock before calling :c:func:`mpsl_init`, the clock configuration option :kconfig:option:`CONFIG_SYSTEM_CLOCK_NO_WAIT` has no effect.

Reviewer 3: ```suggestion
  If the application starts the Low Frequency Clock before calling the :c:func:`mpsl_init` function, the clock configuration option :kconfig:option:`CONFIG_SYSTEM_CLOCK_NO_WAIT` has no effect.
```

The issues in this section are related to :ref:`nrfxlib:softdevice_controller`.
In addition to the known issues listed here, see also :ref:`softdevice_controller_limitations` for permanent limitations.

.. rst-class:: v2-9-0 v2-8-0

> DRGN-23776: Sending CIS packets with invalid MIC

Reviewer 14: This issue is fixed and not present in v2.9

Reviewer 14: We thought we had fixed it but the same issue occurs sporadically. Could you add 2.9 to the affected version for DRGN-23776 please?

The issues in this section are related to :ref:`nrfxlib:nrf_802154`.
In addition to the known issues listed here, see also :ref:`802.15.4 Radio driver limitations <nrf_802154_limitations>` for permanent limitations.

.. rst-class:: v2-9-0 v2-8-0

KRKNWK-19574: nRF 802.15.4 Radio Driver stuck in ``RADIO RXDISABLE`` state

>   The nRF 802.15.4 Radio Driver might stuck in the function :c:func:`wait_until_radio_disabled` while waiting for the radio peripheral transition from state ``RXDISABLE`` to ``DISABLED``.

Reviewer 3: ```suggestion
  The nRF 802.15.4 Radio Driver might get stuck in the function :c:func:`wait_until_radio_disabled` while waiting for the radio peripheral transition from state ``RXDISABLE`` to ``DISABLED``.
```
